From 076d59ffdb40456eb21533d89633e59a4f734aed Mon Sep 17 00:00:00 2001
From: Jiangjin Wang <kaymw@aosc.io>
Date: Mon, 2 Feb 2026 22:14:20 -0800
Subject: [PATCH 1/8] AOSCOS: Disable LASX by default for libyuv

We are starting to have powerful mobile and embedded chips that
lack LASX capabilities, namely 3B6000M and 2K3000.

Co-Authored-By: Xuan Chen <henry.chen@oss.cipunited.com>
---
 src/media/libyuv/libyuv/libyuv-gyp.patch | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)
 create mode 100644 src/media/libyuv/libyuv/libyuv-gyp.patch

diff --git a/src/media/libyuv/libyuv/libyuv-gyp.patch b/src/media/libyuv/libyuv/libyuv-gyp.patch
new file mode 100644
index 000000000..53a8a4f09
--- /dev/null
+++ b/src/media/libyuv/libyuv/libyuv-gyp.patch
@@ -0,0 +1,16 @@
+diff --git a/media/libyuv/libyuv/libyuv.gyp b/media/libyuv/libyuv/libyuv.gyp
+index 2ae75fefd8a0..41aaf319abfe 100644
+--- a/media/libyuv/libyuv/libyuv.gyp
++++ b/media/libyuv/libyuv/libyuv.gyp
+@@ -26,6 +26,11 @@
+     'clang%': 0,
+     # Link-Time Optimizations.
+     'use_lto%': 0,
++    # Disable LASX on LoongArch by default (for 2K3000/3B6000M).
++    'loong_lasx%': 0,
++    # Enable LSX on LoongArch by default. Has no effect if loong_lasx is
++    # enabled because LASX implies LSX according to the architecture specs.
++    'loong_lsx%': 1,
+     'build_neon': 0,
+     'conditions': [
+        ['(target_arch == "armv7" or target_arch == "armv7s" or \
-- 
2.52.0

